File: qualia-tempo-prototype/frontend/src/services/interfaces/IConfigurationService.ts
/**
 * QUALIA.CODE v1.1 - IConfigurationService Interface
 * 
 * ARCHITECTURAL MISSION: Generic, type-safe configuration provider.
 * NO knowledge of specific service configuration structures.
 * Eliminates god object anti-pattern through decoupled design.
 */

import type { FullGameConfig } from '../../types/config';

/**
 * Minimal, powerful configuration service interface.
 * 
 * PRINCIPLE: ConfigurationService is a generic configuration provider.
 * It does NOT define configuration contracts - those belong to individual services.
 * It provides type-safe access to externally-defined configuration structures.
 */
export interface IConfigurationService {
  /**
   * Load configuration from external YAML files.
   * @returns Promise that resolves with the complete loaded configuration
   */
  loadConfig(): Promise<FullGameConfig>;

  /**
   * Check if configuration is loaded.
   * @returns True if configuration is loaded, false otherwise
   */
  isLoaded(): boolean;

  /**
   * Reload configuration from external sources.
   * @returns Promise that resolves when configuration is reloaded
   */
  reload(): Promise<void>;
}

File: qualia-tempo-prototype/backend/services/interfaces/IConfigurationService.py
# QUALIA.CODE v1.1 - IConfigurationService Interface
# Interface for configuration management service

from typing import Protocol, Any, TypeVar, Type

T = TypeVar('T')


class IConfigurationService(Protocol):
    """Interface for configuration management service."""

    async def load_config(self, config_name: str) -> dict[str, Any]:
        """Load configuration from YAML file."""
        ...

    def get[T](self, key: str, config_type: Type[T]) -> T:
        """Get typed configuration value."""
        ...

    def get_raw(self, key: str, default: Any = None) -> Any:
        """Get raw configuration value."""
        ...

    def reload(self) -> None:
        """Reload all configurations."""
        ...
